packages/flutter/lib/src/gestures/drag_details.dart

```diff
@@ -6,9 +6,36 @@
 import 'dart:ui' show Offset, PointerDeviceKind;
 
 import 'package:flutter/foundation.dart';
+import 'package:flutter/gestures.dart';
 
 import 'velocity_tracker.dart';
 
+
+/// Details object that contains the [PointerEvent] that caused it.
+///
+/// See also:
+///  * [DragDownDetails], which takes a [PointerDownEvent].
+///  * [DragStartDetails], which takes a [PointerDownEvent].
+///  * [DragUpdateDetails], which takes a [PointerMoveEvent].
+///  * [DragEndDetails], the details for [PointerUpEvent].
+abstract class DragDetails {
```

Comment:
might want to type this
```suggestion
abstract class DragDetails<T extends PointerEvent> {
```